### PR TITLE
Clean up around PathFilter str/bytes boundary

### PR DESCRIPTION
**Pull request by foolip**
Part of https://github.com/web-platform-tests/wpt/issues/28776.

### CODE REVIEW

**Comment by foolip** [comment]
@gsnedders perhaps you have thoughts on this one. Do you think that the gitignore code should continue to operate on bytes going forward, or was that just most convenient at the time this was last touched? If `PathFilter` operated on strings, then there would be no str/bytes boundary here to maintain, I think.

File: tools/lint/lint.py [hunk 1]
```diff
@@ -116,7 +115,7 @@ def all_filesystem_paths(repo_root, subdir=None):
             if subdir:
                 path = os.path.join(subdir_str, path)
             assert not os.path.isabs(path), path
-            yield ensure_text(path)
+            yield path.decode("utf8")
```
Comment (annevk) [inline tools/lint/lint.py:118]: This is wrong on certain platforms, but perhaps we don't have to care about such files?

Comment (annevk) [reply tools/lint/lint.py:118]: Maybe file an issue on that?

**Review by annevk — APPROVED** [review]

File: tools/lint/lint.py (hunk 1 above)
Comment (foolip) [reply tools/lint/lint.py:118]: I've filed https://github.com/web-platform-tests/wpt/issues/28933.

**Pull request merged**